### ISSUE TITLE
Support for named groups

### ISSUE THREAD

**Issue by JesusTheHun** [davidwebca/tailwindcss-group-variants#1]
Another plugin [tailwindcss-names-groups](https://github.com/ErickTamayo/tailwindcss-named-groups) allow to create several based group names, it would be awesome to be able to combine both !

So with the other plugin you create `group-a` and `group-b` and with your plugin you can expand those to `group-a-first` and `group-b-first` !

Maybe this will require to include his code into your project, but that's the magic of FOOS !

**Comment by davidwebca** [comment]
Hi! Sorry for the delay. Actually, if I understand right, I think my plugin already does what you want. But can you give me a code example of what it would look like (html / css)?

As far as I understand, my plugin already supports naming groups too since 'group' in the example I provide is totally customizable. The example I provide later on in the readme shows this in action : https://codepen.io/davidwebca/pen/YzWdLqz

Where "accordion" is the grouping, "open" is the utility suffix and ".open" is the grouping suffix. In the end, those are just strings that are concatenated together which means you can create anything with it.  I don't know that it makes sense as it could be way more general and applicable to other things than accordions by keeping general names like "group", but it's just an example after all hehe.

TL;DR The name of the variant is totally arbitrary, it's just a variable and only its parameters are used and concatenated to create the end result CSS.

```js
module.exports = {
  variants: {
    margin: ['responsive', 'foo'] // foo is only used here to add the variant to certain css properties
  },
  // foo is arbitrary, only bar, fuz and -boz will actually exist in the resulting css
  groupVariants: {
    'foo': ['bar', 'fuz', '-boz']
  },
  plugins: [
    require('tailwindcss-group-variants'),
  ]
}
```
Ex: no foo!
```css
.bar-boz .bar-fuz\:m-0{
  margin: 0;
}
```

**Comment by JesusTheHun** [comment]
So in order to have variants `first`and `last` for `group-a` and `group-b` I would need to write : 

```js
'group-a-first': ['group-a', 'first', ':first-child'],
'group-b-first': ['group-b', 'first', ':first-child'],
'group-a-last': ['group-a', 'last', ':last-child'],
'group-b-last': ['group-b', 'last', ':last-child'],
```

Are default group variants such as `hover` generated ? The idea of `group-first` is to create a group variant while the idea of `group-a` is to create a group context. 
It feels a bit unnatural to mix them both. 
Also how to I proceed if I want to create a group name without additional variants ? I have to use the other plugin, right ?

**Comment by davidwebca** [comment]
Mmm I created this plugin to solve a problem where I needed to name my groups and active classes or use existing variants without conflicting with the existing behavior. If you would need to combine them with other variants, you would have to do it manually in my plugin's config declaration through something like this : 

```js
module.exports = {
  purge: [],
  darkMode: false,
  theme: {
    extend: {},
  },
  variants: {
    extend: {
        display: ['first-accordion-active']
    },
  },
  groupVariants: {
    'first-accordion-active': ['accordion', 'content', '.active:first-child']
  },
  plugins: [
      require('tailwindcss-group-variants')
  ],
}
```

```css
.accordion.active:first-child .accordion-content\:block {
  display: block;
}

.accordion.active:first-child .accordion-content\:inline-block {
  display: inline-block;
}

.accordion.active:first-child .accordion-content\:inline {
  display: inline;
}

.accordion.active:first-child .accordion-content\:flex {
  display: flex;
}

.accordion.active:first-child .accordion-content\:inline-flex {
  display: inline-flex;
}
/* etc. */
```

Do note though that responsive variants (and some other, off the top of my memory) are stackable by default, which means responsive variants will get generated by default.

```css
.accordion.active:first-child .lg\:accordion-content\:block {
    display: block;
  }

  .accordion.active:first-child .lg\:accordion-content\:inline-block {
    display: inline-block;
  }

  .accordion.active:first-child .lg\:accordion-content\:inline {
    display: inline;
  }
  ```

**Comment by davidwebca** [comment]
Hi! Just wanted to close the issue in case someone stumbles onto this, but I'm basically going to retire this plugin very soon. The release of Tailwind 3 makes it even more easy to create variants and the JIT engine allows us to stack them pretty easily. Here's an example of custom named groups if anyone wants to fiddle with it: https://play.tailwindcss.com/Gp4tfEXDYU?file=config

```js
const plugin = require('tailwindcss/plugin')

module.exports = {
  theme: {
    extend: {
      // ...
    },
  },
  plugins: [
    plugin(({addVariant}) => {
        addVariant('custom-group-name-hover', ':merge(.custom-group-name):hover &')
        addVariant('custom-group-name-first', ':merge(.custom-group-name):first-child &')
        // :merge() is important to merge all variant pseudos on that selector and not the child
        // this is inspirted by the core .group plugin https://github.com/tailwindlabs/tailwindcss/blob/master/src/corePlugins.js#L107
    })
  ],
}
````
